src/configuration.ts

```diff
@@ -1,9 +1,12 @@
+import globalAxios, { AxiosInstance, AxiosRequestConfig } from 'axios';
+
 export type MaestroSupportedNetworks = 'Mainnet' | 'Preprod' | 'Preview';
 
 export interface ConfigurationParameters {
-    apiKey: string;
-    network: MaestroSupportedNetworks;
-    baseOptions?: any;
+    readonly apiKey: string;
+    readonly network: MaestroSupportedNetworks;
+    readonly baseOptions?: AxiosRequestConfig;
```

Comment:
I added the appropriate type here. But this should probably be removed altogether if we allow user to pass the axios instance.
P.s. I didn't do it in this PR because 1) it's a breaking change and 2) it would have made it harder to review.